stuff/library/shaders/programs/HSLBlendGPU.frag

```diff
@@ -114,13 +115,21 @@ void main( void )
   // Perform blending
   if (fg_alpha > 0.0 && bg_alpha > 0.0) {
     vec3 o_pix = SetLumSat(bhue ? fg_pix : bg_pix, bsat ? fg_pix : bg_pix, blum ? fg_pix : bg_pix);
-    gl_FragColor.rgb = mix(bg_pix, o_pix, balpha);
+    if (matte) {
+      gl_FragColor.rgb = mix(bg_pix, o_pix, balpha * fg_frag.a);
+    } else {
+      gl_FragColor.rgb = mix(bg_pix, o_pix, balpha);
+    }
   } else if (fg_alpha > 0.0) {
     gl_FragColor.rgb = fg_pix;
   } else {
     gl_FragColor.rgb = bg_pix;
   }
 
   // Premultiplication
-  gl_FragColor.rgb *= gl_FragColor.a;
+  if (matte) {
+    gl_FragColor.rgb = gl_FragColor.rgb * gl_FragColor.a;
+  } else {
+    gl_FragColor.rgb *= gl_FragColor.a;
+  }
```

Comment:
This code is completely redundant and doesn't change the implementation in any way.

`gl_FragColor.rgb *= gl_FragColor.a;` is the shorthand of `gl_FragColor.rgb = gl_FragColor.rgb * gl_FragColor.a;`